Sources/SwiftGenKit/Parsers/Strings/PlaceholderType.swift

```diff
@@ -118,4 +137,40 @@ extension Strings.PlaceholderType {
     }
     return list
   }
+
+  /// Creates an array of `PlaceholderType` from an array of format chars and their optional positional specifier
+  ///
+  /// This method will normalize placeholders if a positional specifier is missing.
+  /// E.g. "%2$d %3$d" will result in `[.int, .int]`
+  /// This behavior is closer to the how Foundation handles strings with format specifiers during runtime.
+  ///
+  /// - Parameter chars: An array of format chars and their optional positional specifier
+  /// - Throws: `Strings.ParserError.invalidPlaceholder` in case a `PlaceholderType` would be overwritten
+  /// - Returns: An array of `PlaceholderType`
+  static func placeholderTypesFromCharsNormalized(_ chars: [(String, Int?)]) throws -> [Strings.PlaceholderType] {
+    var list = [Int: Strings.PlaceholderType]()
+    var nextNonPositional = 1
+
+    for (str, pos) in chars {
+      guard let char = str.first, let placeholderType = Strings.PlaceholderType(formatChar: char) else { continue }
+      let insertionPos: Int
+      if let pos = pos {
+        insertionPos = pos
+      } else {
+        insertionPos = nextNonPositional
+        nextNonPositional += 1
+      }
+      guard insertionPos > 0 else { continue }
+
+      if let existingEntry = list[insertionPos] {
+        throw Strings.ParserError.invalidPlaceholder(previous: existingEntry, new: placeholderType)
+      } else {
+        list[insertionPos] = placeholderType
+      }
+    }
+
+    return list
+      .sorted { $0.0 < $1.0 } // Sort by key, i.e. the positional value
+      .map { $0.value }
+  }
```

Comment:
Mutualized the implementations into one method in 22bb34397fee4fd0bcf5c295160b1f098898ad07 then decided to get rid of the parameter and use the "mutualized" behavior in all cases in 7a4e98b4f893a3791e8c6fca05ded592919c9e0d

After all, since we double-checked that what you implemented (normalized behavior) _is_ how Foundation behaves, better use it everywhere.

I also added a separate corresponding entry for this fix in the "Bug Fixes" section of the CHANGELOG.